qtpy/QtGui.py

```diff
@@ -8,7 +8,30 @@
 
 """Provides QtGui classes and functions."""
 
-from . import PYQT6, PYQT5, PYSIDE2, PYSIDE6
+from . import PYQT6, PYQT5, PYSIDE2, PYSIDE6, QtModuleNotInstalledError
+from .utils import _getattr_missing_optional_dep
+
+
+_missing_optional_names = {}
+
+_QTOPENGL_NAMES = {
+    'QOpenGLBuffer',
+    'QOpenGLContext',
+    'QOpenGLContextGroup',
+    'QOpenGLDebugLogger',
+    'QOpenGLDebugMessage',
+    'QOpenGLFramebufferObject',
+    'QOpenGLFramebufferObjectFormat',
+    'QOpenGLPixelTransferOptions',
+    'QOpenGLShader',
+    'QOpenGLShaderProgram',
+    'QOpenGLTexture',
+    'QOpenGLTextureBlitter',
+    'QOpenGLVersionProfile',
+    'QOpenGLVertexArrayObject',
+    'QOpenGLWindow',
+}
+
```

Comment:
```suggestion

def __getattr__(name):
    """Custom getattr to chain and wrap errors due to missing optional deps."""
    raise getattr_missing_optional_dep(
        name, module_name=__name__, optional_names=_missing_optional_names)

```

I moved this function here because I think it'd be easier to spot for future maintenance than if it's placed at the bottom of the file.